### PR TITLE
Turn negetive numbers to 0

### PR DESCRIPTION
**Pull request by EyalMecilati**

### CODE REVIEW

File: src/comps/graphs/graphParts.tsx [hunk 1]
```diff
@@ -83,3 +84,18 @@ export function getName(name: string) {
   const arr = name.split(" ").slice(0, 2);
   return arr.join(" ");
 }
+
+export  const getnoNegitiveNumData = (companyDataArr:Array<Object>) => {
+  const newArrWithNoNegitiveNum = []
```
Comment (alissaVrk) [inline src/comps/graphs/graphParts.tsx:89]: few things:
1. there is no need to go over all the data fields only the params
2. you are copying the object each time.. it isn't good here

File: src/comps/graphs/graphParts.test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { getnoNegitiveNumData } from "./graphParts";
+
+const objectArrAllNegetive = [
```
Comment (alissaVrk) [inline src/comps/graphs/graphParts.test.js:3]: it's very nice there are tests
if the input is an array, you should have a test for more than 1 item and for empty

File: src/comps/graphs/useGraphData.ts [hunk 3]
```diff
@@ -9,8 +9,10 @@ export default function useGraphData<T, K extends keyof T>(
   const [graphData, setGraphData] = useState<Array<T>>([]);
 
   useEffect(() => {
+    const validData = data.filter((d) => params.every((param) => isNumber(d[param.field])));
+    const noNegitiveNumData = getnoNegitiveNumData(validData)
     setGraphData(
-      data.filter((d) => params.every((param) => isNumber(d[param.field])))
```
Comment (alissaVrk) [inline src/comps/graphs/useGraphData.ts:13]: we still need to filter out items with missing params

**Review by alissaVrk — CHANGES_REQUESTED** [review]

File: src/comps/graphs/useGraphData.ts (hunk 3 above)
Comment (EyalMecilati) [reply src/comps/graphs/useGraphData.ts:13]: we are, if Im not mistaken? 
const validData = data.filter((d) => params.every((param) => isNumber(d[param.field])));
const noNegitiveNumData = getnoNegitiveNumData(validData)

Comment (alissaVrk) [reply src/comps/graphs/useGraphData.ts:13]: I see the first line deleted

Comment (EyalMecilati) [reply src/comps/graphs/useGraphData.ts:13]: its not

File: src/comps/graphs/graphParts.tsx [hunk 4]
```diff
@@ -83,3 +83,17 @@ export function getName(name: string) {
...
   return arr.join(" ");
 }
+
+type Obj = { [key: string]: any };
+export  const getnoNegitiveNumData = (companyDataArr:Array<Object>,paramsFields:string[]) => {
+  const newArrWithNoNegitiveNum = []
+  for(let companyData of companyDataArr){
+      let noNnegativeNumDataObj:Obj = companyData;
+      for(let field of paramsFields){
+        noNnegativeNumDataObj[field] =  noNnegativeNumDataObj[field] < 0 ? 0 : noNnegativeNumDataObj[field];
```
Comment (alissaVrk) [inline src/comps/graphs/graphParts.tsx:93]: you are changing the passed in company data - this is not good
might have unknown side effects

File: src/comps/graphs/useGraphData.ts [hunk 5]
```diff
@@ -9,8 +9,11 @@ export default function useGraphData<T, K extends keyof T>(
   const [graphData, setGraphData] = useState<Array<T>>([]);
 
   useEffect(() => {
+    const validData = data.filter((d) => params.every((param) => isNumber(d[param.field])));
+    const paramsFieldsArr = params[1] ? [params[0].field.toString(),params[1].field.toString()] : [params[0].field.toString()]
```
Comment (alissaVrk) [inline src/comps/graphs/useGraphData.ts:13]: `params.map` will be easier to read I think

**Review by alissaVrk — CHANGES_REQUESTED** [review]

File: src/comps/graphs/graphParts.tsx [hunk 6]
```diff
@@ -84,3 +84,18 @@ export function getName(name: string) {
...
 }
+
+
+export function getnoNegitiveNumData<T extends { name: string }, K extends keyof T>(companyDataArr:Array<T>, paramsFields: K[]) {
+  const newArrWithNoNegitiveNum = [];
+  for(let companyData of companyDataArr){
+      let noNnegativeNumDataObj: Partial<T> = {};
+      for(let field of paramsFields){
+        noNnegativeNumDataObj[field] =  ((companyData[field] as unknown as number) < 0 ? 0 : companyData[field]) as T[K];
+        noNnegativeNumDataObj['name'] = companyData['name'];
```
Comment (alissaVrk) [inline src/comps/graphs/graphParts.tsx:95]: this is should be outside of the for loop :)

**Review by alissaVrk — DISMISSED** [review]

**Review by alissaVrk — APPROVED** [review]

**Pull request merged**